### PR TITLE
Upgrade to JUnit 5.3.1

### PR DESCRIPTION
**Pull request by mszarlinski**

### CODE REVIEW

File: spring-petclinic-vets-service/pom.xml [hunk 1]
```diff
@@ -42,11 +46,13 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
+            <exclusions>
```
Comment (arey) [inline spring-petclinic-vets-service/pom.xml]: In order to exclude the junit:junit dependency a single time, we could centralize the exclusion into the <dependenciesManagement> section of the parent POM

File: pom.xml [hunk 2]
```diff
@@ -27,10 +27,14 @@
     </modules>
 
     <properties>
-        <assertj.version>3.9.0</assertj.version>
+        <java.version>1.8</java.version>
+        <junit.version>5.3.1</junit.version>
```
Comment (arey) [inline pom.xml]: Do we still need the junit dependency? I'm note sure.

File: pom.xml [hunk 3]
```diff
@@ -55,6 +59,18 @@
                 <scope>import</scope>
             </dependency>
 
+            <dependency>
```
Comment (arey) [inline pom.xml]: junit-jupiter depencies are already declared into the Spring Boot 2.0.4 BOM 
```
        <junit-jupiter.version>5.1.1</junit-jupiter.version>
        <junit-platform.version>1.1.0</junit-platform.version>

...
            <dependency>
                <groupId>org.junit.jupiter</groupId>
                <artifactId>junit-jupiter-api</artifactId>
                <version>${junit-jupiter.version}</version>
            </dependency>
            <dependency>
                <groupId>org.junit.jupiter</groupId>
                <artifactId>junit-jupiter-engine</artifactId>
                <version>${junit-jupiter.version}</version>
            </dependency>
            <dependency>
                <groupId>org.junit.jupiter</groupId>
                <artifactId>junit-jupiter-params</artifactId>
                <version>${junit-jupiter.version}</version>
            </dependency>
```

File: pom.xml (hunk 2 above)
Comment (mszarlinski) [reply pom.xml]: I forgot versions are already provided from parent POM

**Pull request merged**